src/main/java/org/opensearch/knn/quantization/models/quantizationState/OneBitScalarQuantizationState.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensearch.Version;
+import org.opensearch.knn.quantization.models.quantizationParams.ScalarQuantizationParams;
+import org.opensearch.knn.quantization.util.VersionContext;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+/**
+ * OneBitScalarQuantizationState represents the state of one-bit scalar quantization,
+ * including the mean values used for quantization.
+ */
+@Getter
+@NoArgsConstructor // No-argument constructor for deserialization
+@AllArgsConstructor
+public final class OneBitScalarQuantizationState implements QuantizationState {
+    private ScalarQuantizationParams quantizationParams;
+    /**
+     * Mean thresholds used in the quantization process.
+     * Each threshold value corresponds to a dimension of the vector being quantized.
+     *
+     * Example:
+     * If we have a vector [1.2, 3.4, 5.6] and mean thresholds [2.0, 3.0, 4.0],
+     * the quantization process will be:
+     * - 1.2 < 2.0, so the first bit is 0
+     * - 3.4 > 3.0, so the second bit is 1
+     * - 5.6 > 4.0, so the third bit is 1
+     * The quantized vector will be [0, 1, 1].
+     */
+    private float[] meanThresholds;
+    private static final long serialVersionUID = 1L; // Version ID for serialization
+
+    @Override
+    public ScalarQuantizationParams getQuantizationParams() {
+        return quantizationParams;
+    }
+
+    /**
+     * This method is responsible for writing the state of the OneBitScalarQuantizationState object to an external output.
+     * It includes versioning information to ensure compatibility between different versions of the serialized object.
+     *
+     * <p>Versioning is managed using the {@link VersionContext} class. This allows other classes that are serialized
```

Comment:
Are you sure we need this VersionContext? Seems like it could have some issues and rather not need to deal with it.

Reply:
There are two options . Each class writes version, which takes more memory , that's why we are writing version once , now when we read the version , we need to pass to child class for desealize according to version. either I pass that version in method to accomplish that and to do that I will write something like 

ReadExternal( In, version) which is not given by Externaible interface so it will overide default implementation , which creates much confusion. ThreadLocal is local to caller thread , I am not forcesing any potential issue. I can revisit in future if we need to.

Reply:
Removed it .